Formula/llvm@9.rb

```diff
@@ -108,22 +125,53 @@ def install
       -DWITH_POLLY=ON
       -DFFI_INCLUDE_DIR=#{Formula["libffi"].opt_lib}/libffi-#{Formula["libffi"].version}/include
       -DFFI_LIBRARY_DIR=#{Formula["libffi"].opt_lib}
-      -DLLVM_CREATE_XCODE_TOOLCHAIN=ON
       -DLLDB_USE_SYSTEM_DEBUGSERVER=ON
       -DLLDB_DISABLE_PYTHON=1
       -DLIBOMP_INSTALL_ALIASES=OFF
     ]
 
+    on_macos do
+      args << "-DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON" if MacOS.version <= :mojave
+      args << "-DLLVM_CREATE_XCODE_TOOLCHAIN=ON"
+      args << "-DLLVM_ENABLE_LIBCXX=ON"
+      args << "-DDARWIN_osx_ARCHS=x86_64;x86_64h"
+    end
+
+    on_linux do
+      args << "-DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON"
+      args << "-DLLVM_CREATE_XCODE_TOOLCHAIN=OFF"
+      args << "-DLLVM_ENABLE_LIBCXX=OFF"
+      args << "-DCLANG_DEFAULT_CXX_STDLIB=libstdc++"
+
+      # Enable llvm gold plugin for LTO
+      args << "-DLLVM_BINUTILS_INCDIR=#{Formula["binutils"].opt_include}"
+
+      args += %w[
+        -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF
+        -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+
+        -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON
+        -DLIBCXX_STATICALLY_LINK_ABI_IN_SHARED_LIBRARY=OFF
+        -DLIBCXX_STATICALLY_LINK_ABI_IN_STATIC_LIBRARY=ON
+        -DLIBCXX_USE_COMPILER_RT=ON
+
+        -DLIBCXXABI_ENABLE_STATIC_UNWINDER=ON
+        -DLIBCXXABI_STATICALLY_LINK_UNWINDER_IN_SHARED_LIBRARY=OFF
+        -DLIBCXXABI_STATICALLY_LINK_UNWINDER_IN_STATIC_LIBRARY=ON
+        -DLIBCXXABI_USE_COMPILER_RT=ON
+        -DLIBCXXABI_USE_LLVM_UNWINDER=ON
+
+        -DLIBUNWIND_USE_COMPILER_RT=ON
+      ]
+    end
+
     mkdir "build" do
-      if MacOS.version >= :mojave
-        sdk_path = MacOS::CLT.installed? ? "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" : MacOS.sdk_path
-        args << "-DDEFAULT_SYSROOT=#{sdk_path}"
-      end
+      args << "-DDEFAULT_SYSROOT=#{opt_prefix}" if MacOS.version >= :big_sur
```

Comment:
Looks like this didn't work. I think we should just not pass `DEFAULT_SYSROOT` at all, since that makes things very broken (can't find `libSystem` on Big Sur).

Reply:
Unfortunately, I changed that because it didn't fix the problem when I didn't set it either.  Though setting this clearly made it worse.  The real problem here is that, even if we don't set `DEFAULT_SYSROOT`, it still somehow ends up adding the include paths for the SDK anyway.  I'm not sure where that is happening, but if we can stop the SDK paths from being added by default, the tests should pass.

Reply:
That's from environment variables being set in the `test` block. Try
```ruby
ENV.remove_macosxsdk if MacOS.version >= :big_sur
```
We might also need [`ENV.remove_cc_etc`](https://rubydoc.brew.sh/SharedEnvExtension.html#remove_cc_etc-instance_method), but I think [`ENV.remove_macosxsdk`](https://rubydoc.brew.sh/Stdenv.html#remove_macosxsdk-instance_method) removes everything that `#remove_cc_etc` does and then some.

At any rate, this never passed tests on Big Sur, so I think it should be fine even if we can't get it to work there.

Reply:
Ah, in that case this I'll not set `DEFAULT_SYSROOT`, and I'll add that line in the test block.  

>At any rate, this never passed tests on Big Sur, so I think it should be fine even if we can't get it to work there.

I agree that we should merge as is if we still can't get it to pass tests.  Fixing this in Big Sur was never the original goal of this PR but came "for free" because the fix for Catalina happened to let the bottle build.  Though I plan to follow up on the issue in a separate PR after the other LLVM versions are taken care of.